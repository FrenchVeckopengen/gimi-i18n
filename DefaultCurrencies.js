// @flow
export default {
  'SE': 'SEK', // Swende
  'NO': 'NOK', // Norway
  'DK': 'DKK', // Denmark
  'GB': 'GBP', // Britain
  'US': 'USD', // US
  'FI': 'EUR', // Finland
  'SV': 'USD', // El salvadore ?
  'PA': 'USD', // Panama ?
  'EC': 'USD', // Equador ?
  'GU': 'USD',
  'PW': 'USD',
  'PR': 'USD',
  'ZW': 'USD',
  'AU': 'AUD', // Australia
  'CA': 'CAD',
  'NZ': 'NZD',
  'IN': 'INR',
  'ZA': 'ZAR',
  'IT': 'EUR', // Italy
  'BE': 'EUR', // Belgium
  'ES': 'EUR', // Spain
  'DE': 'EUR', // Germany,
  'EE': 'EUR', // Estonia
  'IS': 'ISK', // Iceland
  'TH': 'THB',
  'ID': 'IDR' // Indonesia
}
