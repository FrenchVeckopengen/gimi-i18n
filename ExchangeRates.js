// @flow
export default {
  'SEK': 1,
  'DKK': 0.8,
  'NOK': 1,
  'EUR': 0.11,
  'GBP': 0.09,
  'USD': 0.12,
  'AUD': 0.15,
  'CAD': 0.15,
  'NZD': 0.16,
  'INR': 7.9,
  'ZAR': 1.59,
  'NAD': 1.59,
  'BWP': 1.22,
  'ZWD': 42.27,
  'SZL': 1.58,
  'LSL': 1.58,
  'ZMF': 1.19,
  'TZS': 255.7,
  'KES': 11.85,
  'UGX': 393,
  'NGN': 37.2,
  'XAF': 69.15,
  'GHS': 0.46,
  'LRD': 10.51,
  'SLL': 460.37,
  'GMD': 5,
  'SCR': 1.56,
  'MUR': 4.11,
  'SHP': 0.09,
  'MGA': 347.75,
  'DJF': 20.75,
  'JPY': 12,
  'CNY': 0.79,
  'RUB': 7.6,
  'MXN': 2.16,
  'PKR': 12.21,
  'BRL': 0.37,
  'ARS': 1.72,
  'AED': 0.43,
  'HKD': 0.91,
  'SGD': 0.16,
  'CHF': 0.12,
  'ISK': 12.68
}
