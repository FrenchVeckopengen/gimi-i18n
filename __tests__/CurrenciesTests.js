/* eslint flow-header/flow-header: 0 */
import {getSupportedCurrencyInfos, getDefaultCurrencyCode} from '../index'
import DefaultCurrencies from '../DefaultCurrencies'
import ExchangeRates from '../ExchangeRates'
import 'babel-polyfill'
import Decimal from 'decimal.js-light'

jest.disableAutomock()

describe('Currencies', () => {
  it('it should return supported supportedCurrencyCodes', () => {
    expect(getSupportedCurrencyInfos()).not.toEqual(undefined)
    expect(getSupportedCurrencyInfos().length).toEqual(16)
    expect(getSupportedCurrencyInfos()).toMatchSnapshot()
  })

  it('it should return default currency', () => {
    expect(getDefaultCurrencyCode('FI')).toEqual('EUR')
    expect(getDefaultCurrencyCode('NO')).toEqual('NOK')
    expect(getDefaultCurrencyCode('GB')).toEqual('GBP')
    expect(getDefaultCurrencyCode('SE')).toEqual('SEK')
    expect(getDefaultCurrencyCode('BE')).toEqual('EUR')
    expect(getDefaultCurrencyCode()).toEqual('EUR')
  })

  it('all currencies in Default CurrencyCodes needs to be in supported Currency Codes', () => {
    Object.values(DefaultCurrencies).forEach((currencyCode) => expect(getSupportedCurrencyInfos().find((currencyInfo) => currencyInfo.code === currencyCode)).toBeDefined())
  })

  Object.keys(DefaultCurrencies).forEach(key => {
    it(`Should be an exchange rate for ${key} (entry found in default currencies)`, () => {
      expect(ExchangeRates[DefaultCurrencies[key]]).toBeDefined()
    })
  })

  Object.keys(getSupportedCurrencyInfos()).forEach(key => {
    it(`All defined currencies should have conversion rate ${ExchangeRates[getSupportedCurrencyInfos()[key].code]} `, () => {
      expect(ExchangeRates[getSupportedCurrencyInfos()[key].code]).toBeDefined()
    })
  })

  Object.keys(getSupportedCurrencyInfos()).forEach(key => {
    it(`All defined currencies should be a positive number ${ExchangeRates[getSupportedCurrencyInfos()[key].code]} `, () => {
      var exchangeRate = new Decimal(ExchangeRates[getSupportedCurrencyInfos()[key].code])

      expect(exchangeRate.isPositive()).toBeTruthy()
      expect(ExchangeRates[getSupportedCurrencyInfos()[key].code]).toBeDefined()
    })
  })
})
