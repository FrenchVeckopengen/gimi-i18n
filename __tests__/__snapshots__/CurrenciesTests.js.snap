// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currencies it should return supported supportedCurrencyCodes 1`] = `
Array [
  Object {
    "code": "SEK",
    "name": "Swedish Krona",
  },
  Object {
    "code": "NOK",
    "name": "Norwegian Krone",
  },
  Object {
    "code": "DKK",
    "name": "Danish Krone",
  },
  Object {
    "code": "GBP",
    "name": "Pound Sterling",
  },
  Object {
    "code": "USD",
    "name": "US Dollar",
  },
  Object {
    "code": "EUR",
    "name": "Euro",
  },
  Object {
    "code": "AUD",
    "name": "Australian Dollar",
  },
  Object {
    "code": "CAD",
    "name": "Canadian Dollar",
  },
  Object {
    "code": "NZD",
    "name": "New Zealand Dollar",
  },
  Object {
    "code": "INR",
    "name": "Indian Rupee",
  },
  Object {
    "code": "ZAR",
    "name": "Rand",
  },
  Object {
    "code": "ISK",
    "name": "Iceland Krona",
  },
  Object {
    "code": "THB",
    "name": "Thai Baht",
  },
  Object {
    "code": "IDR",
    "name": "Indonesian Rupiahs",
  },
  Object {
    "code": "CHF",
    "name": "Swiss Franc",
  },
  Object {
    "code": "COP",
    "name": "Colombian Peso",
  },
]
`;
